### ISSUE TITLE
Scrolloff option

### ISSUE THREAD

**Issue by kalons7** [fman-community/fman#186]
Would ensure that there are always a certain number of lines visible above and below the selected file/folder when scrolling.  Is this possible via a plugin?

See here for Visual Studio Code implementation.
https://marketplace.visualstudio.com/items?itemName=tickleforce.scrolloff

**Comment by mherrmann** [comment]
I'm not familiar with scrolloff, but briefly looked it up now. I'm pretty certain it's not possible to do this with the current plugin API. Do I understand correctly that the goal is for the scrolling to change as the cursor is moved?

**Comment by kalons7** [comment]
Yes, it would keep rows visible above/below the selection while scrolling
(for long folder/file lists), similar to many editors.  XYplorer supports
this (referred to as scroll margin).

On Mon, Jun 26, 2017 at 11:11 PM, Michael Herrmann <notifications@github.com

**Comment by mherrmann** [comment]
Okay, thank you for the clarification!

**Comment by tobia** [comment]
To be clear:
- scrolloff = 0 disables the feature;
- scrolloff ≥ lines / 2 (half or more of the number of entries visible with the current window height) makes the cursor always stay in the center of the screen, and the up/down arrows make the list scroll "behind" the cursor;
- scrolloff = 3 (a reasonable value) makes the cursor behave normally, until it goes up against the window edge; at that point, the list starts scrolling behind the cursor, so as to **always leave 3 lines** between the cursor and the edge of the screen.

This is especially useful when selecting files with Shift+arrow, because without a scroll offset, you never know when to stop selecting. You have to either use the mouse wheel every time the cursor reaches the bottom of the window; or over-select and then de-select the additional files.

Notice that the scroll offset goes down to zero near the beginning or end of the list. That's how you know you have reached the end, because the scroll offset is gone.

You can quickly see how it works by opening Vim on a source code file, set the option with `:set scrolloff=3`+Enter and then scroll around with the arrows. (To quit Vim type Esc+`:q`+Enter.)
